cpp/arcticdb/codec/codec.cpp

```diff
@@ -160,20 +378,18 @@ void calc_encoded_blocks_size(
     const arcticdb::proto::encoding::VariantCodec& codec_opts,
     SizeResult& result
     ) {
-    using BytesTypeDescriptorTag = TypeDescriptorTag<DataTypeTag<DataType::UINT8>, DimensionTag<Dimension::Dim1>>;
-    using BytesEncoder = BlockEncoder<BytesTypeDescriptorTag>;
-
     result.encoded_blocks_bytes_ = static_cast<shape_t>(encoded_blocks_size(in_mem_seg));
     result.uncompressed_bytes_ += result.encoded_blocks_bytes_;
-    result.max_compressed_bytes_ += BytesEncoder::max_compressed_size(codec_opts, TypedBlockData<BytesTypeDescriptorTag>(result.encoded_blocks_bytes_, &result.encoded_blocks_bytes_));
+    result.max_compressed_bytes_ += BytesEncoder<EncodingVersion::V2>::max_compressed_size(codec_opts, result.encoded_blocks_bytes_);
 }
 
+template<EncodingVersion v, typename = std::enable_if_t<v == EncodingVersion::V2>>
```

Comment:
This might also benefit from a tag type, we seem to be using all the different compile-time switches rather than picking one and sticking with it.